kolibri/core/assets/src/views/__tests__/TotalPoints.spec.js

```diff
@@ -0,0 +1,61 @@
+import VueRouter from 'vue-router';
+import { render, screen, fireEvent } from '@testing-library/vue';
+import TotalPoints from '../TotalPoints.vue';
+import '@testing-library/jest-dom';
+
+// Create a mock Vuex store with the required getters and actions
+// This is a helper function to avoid create a new store for each test and not reuse the same object
+const getMockStore = () => {
+  return {
+    getters: {
+      totalPoints: () => 100,
+      currentUserId: () => 'user123',
+      isUserLoggedIn: () => true,
+    },
+    actions: {
+      fetchPoints: jest.fn(),
+    },
+  };
+};
+
+// Helper function to render the component with Vuex store
+const renderComponent = store => {
+  return render(TotalPoints, {
+    store,
+    routes: new VueRouter(),
+  });
+};
+
+describe('TotalPoints', () => {
+  test('renders when user is logged in', async () => {
+    const store = getMockStore();
+    renderComponent(store);
+
+    expect(screen.getByRole('presentation')).toBeInTheDocument();
+    expect(screen.getByText('100')).toBeInTheDocument();
+  });
+
+  test('does not render when user is not logged in', async () => {
+    const store = getMockStore();
+    store.getters.isUserLoggedIn = () => false;
+    renderComponent(store);
+
+    expect(screen.queryByRole('presentation')).not.toBeInTheDocument();
+    expect(screen.queryByText('100')).not.toBeInTheDocument();
+  });
+
+  test('fetchPoints method is called on created', async () => {
+    const store = getMockStore();
+    renderComponent(store);
+
+    expect(store.actions.fetchPoints).toHaveBeenCalledTimes(1);
+  });
+
+  test('tooltip message is displayed correctly when the mouse hovers over the icon', async () => {
+    const store = getMockStore();
+    renderComponent(store);
+
+    await fireEvent.mouseOver(screen.getByRole('presentation'));
```

Comment:
There is something that I dont like here, and it is that if we just read the test, without the context of `getMockStore` we will not know why we are seeing that we "earned 100 points", why `100` I mean.

An unit test, as far as possible, should be simple and self descriptive without needing to switch context and see implementation details of any helper function, usually we could follow the pattern: 
1. Define input data for the test
2. Trigger the action we are testing (in some cases this could be just calling the render function)
3. Assert results

If we just do `const store = getMockStore();` we are not providing visibility about the relevant input data we are testing (in this case the `totalPoints` as it has a direct impact in the results we are asserting).

So, we could do two things.
1. pass props to the mock store to edit relevant input values. So we could do something like:
```javascript
const totalPoints = 100;
const store = getMockStore({ totalPoints });

...

expect(screen.getByText(`You earned ${totalPoints} points`)).toBeInTheDocument();
```
A drawback for this is that if we have multiple modules in the vuex store, the getMockStore function could become a bit messy.

2. Define global objects for props and store (and other input data) that are defined with inital data in the `beforeEach` method, so a test will not affect the input data of another test. And in each test case we can update that store object.

```javascript
let store;
let storeActions;

const renderComponent = () => {
  return render(TotalPoints, {
    store: {
      getters: {
        totalPoints: () => store.totalPoints;
      },
      actions: storeActions
    },
    routes: new VueRouter(),
  });
};

describe('TotalPoints', () => {
  beforeEach(() => {
    store = {
      totalPoints: 100,
      currentUserId: '123',
      ...
    };
    storeActions = {
      fetchPoints: jest.fn()
    }
  });

  test('renders total points correctly if user is logged in', () => {
    store.isUserLoggedIn = true;
    store.totalPoints = 50;

    renderComponent();

    expect(screen.getByText(totalPoints)).toBeInTheDocument();
  });
});
```

I would prefer the latter but I look forward to hearing your comments @EshaanAgg and @MisRob about this pattern, and whether we can standardize it.

Reply:
The second approach looks better to me. It's a bit more boilerplate, and I'm not a fan of the idea of using uninitialized variables, but the increase in the clarity of the tests is profound. The tests in the second part example can be visualized to have three parts:

1. Set the values of the variables and the functions you need (you can assume that all the other ones have meaningful defaults.
2. Call the render function.
3. Make the assertions and test the user interactions.

Reply:
A small doubt and a nit:
```js
test('renders total points correctly if user is logged in', () => {
     // Assuming that the user is logged in by default, should we be redeclaring it here?
     // In Favour: The indent and the pre-conditions are explicitly stated
     // Against: The test-case name already converts the indent, so re-declaration is not really needed
     store.isUserLoggedIn = true; 
 
     // Do we really need this assignment here? (As the value `50` is not special and the default value should be good enough)
     // Moreover we always refer to the same as `store.totalPoints` in the assertions
    store.totalPoints = 50;   

    renderComponent();

    expect(screen.getByText(store.totalPoints)).toBeInTheDocument();
  });
});

```

Reply:
Hmmm I think that if we are Indeed testing something "if user is logged in" I think re-declaring values is valuable as we also show where does these values are coming through.

And about re-declaring the totalPoints I don't have any comments, i think can be any of both, but would prefer to show that we are setting specifically those input values. I would let that there just for visibility.

Reply:
Got it! I will update the test suite to match this pattern! Thanks for the detailed review and the discussion!